model/src/pyrenew/latent/infection_seeding_method.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABCMeta, abstractmethod
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+"""
+Classes for the infection seeding methods
+"""
+
+
+class InfectionSeedMethod(metaclass=ABCMeta):
+    def __init__(self, n_timepoints: int):
+        self.validate(n_timepoints)
+        self.n_timepoints = n_timepoints
+
+    @staticmethod
+    def validate(n_timepoints: int) -> None:
+        assert isinstance(n_timepoints, int), "n_timepoints must be an integer"
+        assert n_timepoints > 0, "n_timepoints must be positive"
+
+    @abstractmethod
+    def seed_infections(self, I0: ArrayLike) -> ArrayLike:
+        pass
+
+
+class SeedInfectionsZeroPad(InfectionSeedMethod):
```

Comment:
This is written to produce the same result as `SeedInfectionsZeroHstack`, but it doesn't work within `sample`. See `scratch/InfectionSeedingProcess_demo.py` for example code.